src/models/transaction.rs

```diff
@@ -0,0 +1,226 @@
+use reth_primitives::{
+    AccessList, AccessListItem, Transaction as PrimitiveTransaction, TransactionKind, TxEip1559, TxEip2930, TxLegacy,
+};
+use reth_rpc_types::Transaction as RpcTransaction;
+
+use super::errors::ConversionError;
+
+pub fn rpc_transaction_to_primitive(rpc_transaction: RpcTransaction) -> Result<PrimitiveTransaction, ConversionError> {
```

Comment:
I think it's more readable to do `reth_primitives::Transaction` and `reth_rpc_types::Transaction` instead of renaming the types.